src/main/java/com/hillel/spring/service/task/TaskServiceImpl.java

```
@@ -0,0 +1,82 @@
+package com.hillel.spring.service.task;
+
+import com.hillel.spring.model.Task;
+import com.hillel.spring.model.TaskStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+
+@Service
+public class TaskServiceImpl implements TaskService {
+    private final List<Task> tasks = new ArrayList<>();
+    private long nextId = 1;
+
+    public List<Task> getAllTasks() {
+        return tasks;
+    }
+@Override
+    public Optional<Task> getTaskById(Long id) {
+        return tasks.stream()
+                .filter(task -> task.getId().equals(id))
+                .findFirst();
+    }
+
+    public Task createTask(Task task) {
+        task.setId(nextId++);
+        tasks.add(task);
+        return task;
+    }
+@Override
+    public Task updateTask(Long id, Task updatedTask) {
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            if (task.getId().equals(id)) {
+                updatedTask.setId(id);
+                tasks.set(i, updatedTask);
+                return updatedTask;
+            }
+        }
+        return null;
```

I would like to throw exception instead of return null